django/api/viewsets/user.py

```diff
@@ -0,0 +1,56 @@
+from rest_framework.decorators import action
+from rest_framework.permissions import AllowAny
+from rest_framework.response import Response
+from rest_framework.viewsets import GenericViewSet
+from api.models.user import User
+from api.serializers.user import UserSerializer, UserSaveSerializer
+
+class UserViewSet(GenericViewSet):
+    """
+    This viewset automatically provides `list`, `create`, `retrieve`,
+    and  `update`  actions.
+    """
+    permission_classes = (AllowAny,)
+    http_method_names = ['get', 'post', 'put', 'patch']
+    queryset = User.objects.all()
+
+    serializer_classes = {
+        'default': UserSerializer,
+        'update': UserSaveSerializer,
+        'create': UserSaveSerializer,
+    }
+
+
+    def get_serializer_class(self):
+        if self.action in list(self.serializer_classes.keys()):
+            return self.serializer_classes[self.action]
+
+        return self.serializer_classes['default']
+
+
+    @action(detail=False)
+    def current(self, request):
+        """
+        Get the current user
+        """
+        user = User.objects.filter(idir=request.user).first()
+        serializer = self.get_serializer(user)
+        return Response(serializer.data)
+    
+    def list(self, request):
```

Comment:
I’m not sure if the “list” action is available since we’re only extending from GenericViewSet; I think you need the ListModelMixin? Also, like you said, checking the user’s permissions can be a reusable thing; a decorator instead of DRF’s permissions framework would also work!

Reply:
@emi-hi Never mind, it looks like GenericViewSet binds GET requests to the list method; specifically, it does this when associating the url with the viewset in urls.py. So we're good!